### ISSUE TITLE
CLI doesn't exit

### ISSUE THREAD

**Issue by Slayer95** [wesnoth/wesnoth#7500]
### Game and System Information

- **Version:**
- 1.16.8
- **Downloaded from:**
- Steam
- **Build info:**
- DNA
- **OS:**
- Windows 7 Ultimate Service Pack 1

### Description of the bug

None of the commands I tried with ``cmd.exe`` ever exited.

E.g.

```
C:\Users\W7>"C:\Program Files (x86)\Steam\SteamApps\common\wesnoth\wesnoth.exe" --version

C:\Users\W7>Battle for Wesnoth v1.16.8 x86_64
Started on Fri Mar 31 09:23:42 2023

Automatically found a possible data directory at C:\Program Files (x86)\Steam\SteamApps\common\wesnoth
Battle for Wesnoth 1.16.8

Library versions:
Boost:             1.80
Lua:               5.4.2
OpenSSL/libcrypto: 1.1.1s (runtime 1.1.1s)
Cairo:             1.17.6 (runtime 1.17.6)
Pango:             1.50.11 (runtime 1.50.11)
SDL:               2.24.1 (runtime 2.24.1)
SDL_image:         2.6.2 (runtime 2.6.2)
SDL_mixer:         2.6.2 (runtime 2.6.2)


Optional features:
Lua console completion:       yes
Win32 notifications back end: yes
```
[Hangs]

### Steps to reproduce the behavior

1. Run ``cmd.exe``
2. ``"C:\Program Files (x86)\Steam\SteamApps\common\wesnoth\wesnoth.exe" --version``
3. Verify that the process never exits.

### Expected behavior

Process should have exited.

### Additional context

Neither `--help`` nor ``--preprocess`` exit either. ``wesnoth(6)`` specifies that ``--help`` and ``--version`` should produce their respective outputs _and_ exit.

**Comment by soliton-** [comment]
Can you try what happens when you use `--wnoconsole`?

**Comment by ProditorMagnus** [comment]
`--wnoconsole` works and provides output https://github.com/ProditorMagnus/WML_tree_tools/commit/c5bfaf5ef82218092573756616183737fd486932

**Comment by ProditorMagnus** [comment]
What you describe as hanging is just cmd user interface issue. You can enter new commands already at that time.

**Comment by Slayer95** [comment]
> What you describe as hanging is just cmd user interface issue. You can enter new commands already at that time.

Uh... So it was just in ``echo off``? Then the issue is very small.

Thanks!

**Comment by ProditorMagnus** [comment]
Specifically, Slayer95 expects
```
C:\Users\W7>"C:\Program Files (x86)\Steam\SteamApps\common\wesnoth\wesnoth.exe" --version
C:\Users\W7>Battle for Wesnoth v1.16.8 x86_64...
```
changed to
```
C:\Users\W7>"C:\Program Files (x86)\Steam\SteamApps\common\wesnoth\wesnoth.exe" --version
Battle for Wesnoth v1.16.8 x86_64...
C:\Users\W7>
```

**Comment by soliton-** [comment]
Does the desired behaviour happen with some combination of `--wnoconsole` and/or `--wnoredirect`?

**Comment by Slayer95** [comment]
> 

> Does the desired behaviour happen with some combination of `--wnoconsole` and/or `--wnoredirect`?

``--wnoconsole`` is sufficient, ``--wnoredirect`` has no influence.

**Comment by Wedge009** [comment]
I've noticed this behaviour before and not only with Wesnoth. It's a minor nuisance, but easy to work past once one recognises what's going on.

**Comment by Slayer95** [comment]
> I've noticed this behaviour before and not only with Wesnoth. It's a minor nuisance, but easy to work past once one recognises what's going on.

It's apparently a common error that arises when the program calls ``echo off`` as opposed to ``@echo off``. Maybe that's also what happens here.

**Comment by CelticMinstrel** [comment]
I don't think that has any relation to this at all. Even if this were a batch script, the effect of doing that would not be what you stated.

**Comment by Pentarctagon** [comment]
I suspect that this is caused by Wesnoth not being built as a console app, and instead writing to CONOUT to get output onto the console. So this may not be solvable.

**Comment by CelticMinstrel** [comment]
Yes, it may not be. It would definitely be nice if it could be fixed, but I don't know of any way to do it without building Wesnoth as a console app.

**Comment by Slayer95** [comment]
Closing since I found out ``cwesnoth.cmd`` exists.

**Comment by Wedge009** [comment]
Interesting - does it do what you expect for a command-line program? I notice it sets `SDL_STDIO_REDIRECT=0`.

**Comment by Slayer95** [comment]
> Interesting - does it do what you expect for a command-line program? I notice it sets `SDL_STDIO_REDIRECT=0`.

I don't see anything obviously wrong in its behavior in 1.16. But redirection does seem broken in 1.17.16. IDK about master.

**Comment by Pentarctagon** [comment]
`SDL_STDIO_REDIRECT` is also an SDL 1.2 option as far as I can find, not something for SDL2.
